src/services/provider/solana/mod.rs

```diff
@@ -35,6 +39,14 @@ pub enum SolanaProviderError {
     RpcError(String),
     #[error("Invalid address: {0}")]
     InvalidAddress(String),
+    #[error("RPC selector error: {0}")]
+    SelectorError(String),
+}
+
+impl From<RpcSelectorError> for SolanaProviderError {
```

Comment:
This could be simplified by defining error as

 #[error("RPC selector error: {0}")]
 SelectorError(#[from] RpcSelectorError),